internal/pkg/service/common/etcdop/op/txn.go

```diff
@@ -26,29 +34,32 @@ import (
 type TxnOp[R any] struct {
 	result     *R
 	client     etcd.KV
-	processors []func(ctx context.Context, r *TxnResult[R])
 	errs       errors.MultiError
-	ifs        []etcd.Cmp
-	thenOps    []Op
-	thenTxnOps []Op // thenTxnOps are separated from thenOps to avoid confusing between Then and Merge
-	mergeOps   []Op
-	elseOps    []Op
+	parts      []txnPart[R]
+	processors []func(ctx context.Context, r *TxnResult[R])
 }
 
 // txnInterface is a marker interface for generic type TxnOp.
 type txnInterface interface {
 	txn()
 }
 
+type txnPartType string
+
+type txnPart[R any] struct {
+	Type    txnPartType
+	If      etcd.Cmp
+	Factory Op
+}
+
 type lowLevelTxn[R any] struct {
-	result      *R
-	client      etcd.KV
-	processors  []func(ctx context.Context, r *TxnResult[R])
-	ifs         []etcd.Cmp
-	thenOps     []etcd.Op
-	elseOps     []etcd.Op
-	thenMappers []MapFn
-	elseMappers []MapFn
+	result     *R
+	client     etcd.KV
+	processors []func(ctx context.Context, r *TxnResult[R])
+	opCounter  map[txnPartType]int
```

Comment:
Why did you add this?

Reply:
looks like telemetry stuff

Reply:
no, no telemetry, ... but better errors messages, let see where it is used:

https://github.com/keboola/keboola-as-code/blob/ea0baab5900336e5496428f2192060c154a572be/internal/pkg/service/common/etcdop/op/txn.go#L275

https://github.com/keboola/keboola-as-code/blob/ea0baab5900336e5496428f2192060c154a572be/internal/pkg/service/common/etcdop/op/txn.go#L292

https://github.com/keboola/keboola-as-code/blob/ea0baab5900336e5496428f2192060c154a572be/internal/pkg/service/common/etcdop/op/txn.go#L297

Reply:
Could we choose different name? I expected that Counter is something tight with telemetry. Wouldn't it be better to have `operations int` ? Looks as number of operations within the transaction. No counter name needed :)

Reply:
I  imroved it a bit, see https://github.com/keboola/keboola-as-code/pull/1771/commits/418425119c8ea23a0a7eccc60d406920b358f8f2

@hosekpeter @Matovidlo please be a little more precise in your review, you could definitely find out why it is there, it is used in one place, thanks :beers: 

I suggest to open IDE with the [final code](https://github.com/keboola/keboola-as-code/pull/1759), to see full picture.

Reply:
Looks alright.